dist/tools/whitespacecheck/check.sh

```diff
@@ -27,8 +27,14 @@ if [ -z "${BRANCH}" ]; then
     BRANCH=$(git rev-list HEAD | tail -n 1)
 fi
 
-git diff --check $(git merge-base ${BRANCH} HEAD)
-if [ $? -gt 0 ]
+git -c core.whitespace="tab-in-indent,tabwidth=4" \
+    diff --check $(git merge-base ${BRANCH} HEAD) -- *.[ch]
+
+RESULT=$?
+
+git -c core.whitespace="trailing-space" \
+    diff --check $(git merge-base ${BRANCH} HEAD)
```

Comment:
> If makes obvious that any line-ending check is done.

Isn't "trailing whitespaces" (as in "any whitespace except `\n`") enough for that?